cli/src/commit_templater.rs

```diff
@@ -689,6 +712,24 @@ fn evaluate_immutable_revset<'repo>(
     Ok(revset)
 }
 
+fn evaluate_user_revset<'repo>(
+    language: &CommitTemplateLanguage<'repo>,
+    span: pest::Span<'_>,
+    revset: String,
+) -> Result<Box<dyn Revset + 'repo>, TemplateParseError> {
+    let repo = language.repo;
+
+    let expression = revset::parse(&revset, &language.revset_parse_context).map_err(|err| {
+        TemplateParseError::expression("Failed to parse revset", span).with_source(err)
+    })?;
+
+    let symbol_resolver = revset_util::default_symbol_resolver(repo, language.id_prefix_context);
```

Comment:
Perhaps, `evaluate_revset(.., expression)` function can be extracted from `evaluate_immutable_revset()`.

Reply:
I extracted `evaluate_revset_expression(...)` but kept `evaluate_immutable_revset(...)` and `evaluate_user_revset(...)`. Should I inline them where they're used?

Reply:
They look good as named functions (because there may be more callers.)